### PR TITLE
Updated documentation to fix #51

### LINKED ISSUE #51: The documentation for LazyListeners refers to container-interop
### Bug Report

|    Q        |   A
|------------ | ------
| Version(s)  | 3.x

#### Summary

This is a documentation bug.
The documentation on Lazy Listerner is refering to `container-interop` which is no longer used and has been replaced by `Psr\Container`.

### PR DESCRIPTION
**Pull request by visto9259**
|    Q          |   A
|-------------- | ------
| Documentation | yes
| Bugfix        | no
| BC Break      | no
| New Feature   | no
| RFC           | no
| QA            | no

### Description

Updated the sections on LazyListerners to reference the use of PSR-11 container instead of container-interop.
Fixes #51

### CODE REVIEW

File: doc/book/lazy-listeners/lazy-listener.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ To create a `LazyListener` instance, you must pass to its constructor:
 - a *definition* of the listener; this is an array defining:
   - a `listener` key, with the name of the listener service to pull from the container.
   - a `method` key, with the name of the method to invoke on the listener instance.
-- a *container*; this is a [container-interop](https://github.com/container-interop/container-interop),
+- a *container*; this is a [PSR-11 container](https://github.com/php-fig/container),
```
Comment (froschdesign) [inline doc/book/lazy-listeners/lazy-listener.md:13]: I would link to the official website.
```suggestion
- a *container*; this is a [PSR-11 container](https://www.php-fig.org/psr/psr-11/),
```

File: composer.json [hunk 2]
```diff
@@ -37,7 +37,7 @@
         "laminas/laminas-stdlib": "^3.18",
         "phpbench/phpbench": "^1.2.15",
         "phpunit/phpunit": "^10.5.5",
-        "psalm/plugin-phpunit": "^0.18.4",
+        "psalm/plugin-phpunit": "^0.19.0",
```
Comment (froschdesign) [inline composer.json:40]: These changes do not belong in this pull request and should be reverted. Also the `composer.lock` file.

**Review by froschdesign — CHANGES_REQUESTED** [review]
Looks good and it is nice to see that you are using [the right branch for bugfixes](https://github.com/laminas/.github/blob/5e7024ad2f9ac03fd187ec0497282b950424c835/CONTRIBUTING.md#what-branch-to-issue-the-pull-request-against). 👍🏻 

Only two small changes need to be made.

**Comment by visto9259** [comment]
Not quite right. I forked 3.14x instead of 3.13.x. That's probably why there are changes to Composer.
